server/services/GameLogic.js

```diff
@@ -55,8 +55,22 @@ class GameLogic {
     /* Return the list of players, hiding their assigned cards */
     getPlayers() { return this.players.map(player => ({ username: player.username, score: player.score })) };
 
+    /**
+     * Shuffles cards in place.
+     * @param {Array} cards
+    */
+    shuffle(cards) {
+        for (let i = cards.length - 1; i > 0; i--) {
+            const j = Math.floor(Math.random() * (i + 1));
+            const x = cards[i];
+            cards[i] = cards[j];
+            cards[j] = x;
+        }
+        return cards;
+    }
+
     /* Return the list of cards played this round, hiding who played them */
-    getPlayedCards() { return this.playedCards.map(card => ({ cardId: card.cardId })).sort(() => .5 - Math.random()) };
+    getPlayedCards() { return this.shuffle(this.playedCards) };
```

Comment:
Can you put back in the `.map(card => ({ cardId: card.cardId }))` so that we hide who played the card from the client